device_calendar/lib/src/models/event.dart

```diff
@@ -0,0 +1,27 @@
+part of device_calendar;
+
+class Event {
+  String title;
+
+  DateTime start;
+  DateTime end;
+
+  Event.fromJson(Map<String, dynamic> json) {
+    if (json == null) {
```

Comment:
actually, i'm not sure this null check should be there as it is. if json was null, the error gets swallowed up and you end up with an event object (given this is a constructor) whose properties have default values. I'd suggest putting in an assert statement to check the json argument https://www.dartlang.org/guides/language/language-tour#assert

Reply:
assert won't work in production code though. Options are to return or to throw exception. You'll need to handle exceptions anyhow, so you might as well throw one

Reply:
"Assert statements have no effect in production code; they’re for development only" - I'm not truly sure what solution you're suggesting here.

Instead of returning the object with default values, maybe we should create a base object with 'isEmpty' flag and set it when json is null? I can't think of a better solution here.

Reply:
@mkieres it's more to ensure that an error gets raised. logically shouldn't happen but if you were debugging and passed in a null then it breaks on assert statement.

@nickrandolph's point is valid though I do see a lot of assert statements being used in the official plugins. whether or not it's the best approach, i'm not sure...probably stick to what Nick is suggesting with letting it return an exception

Reply:
yeh the asserts are being used in a similar way to how we use Debug.WriteLine... they're for development only so they don't accidentally break stuff